docs/configuration/architecture_diff_func.md

```diff
@@ -0,0 +1,129 @@
+# Differences of architectures
+
+The diff function can be used to find the difference between two architectures. Specifically, the diff functions can find data that only exist in each, e.g. node name, pub/sub topic.
+There are four diff functions, which can be divided into functions that compare the entire architecture and functions that compare nodes within the architecture.
+
+ <!-- `diff_node_names()`, `diff_topic_names` are functions that take two architectures and compare them as a whole. The two functions compare all node names in the architecture for the former and all pub/sub topic names in the architecture for the latter.
+
+`diff_node_pubs()`, `diff_node_subs()` and `diff_node_callbacks()` are functions that compare two nodes within an architecture. These are functions that compare two given nodes and return the publish topic name, subscribe topic name, or callback name that is only in each. -->
+
+## How to compare architecture
+
+In CARET, the structure of the target application is described in an architecture file. The actual architecture file can be represented as an object of Architecture class. This section describes the functions `diff_node_names()` and `diff_topic_names()`, which compare the contents of two architecture objects. In `diff_node_names()` and `diff_topic_names()`, the functions compare two architectures and return the node or topic names that only each has. Therefore, by using these functions, what one ARCHITECTURE has but the other does not (node names or topic names) are clear and can be compared.
+One possible use of these functions is to check the changes between the architecture before and after changes.
+
+The diff function is explained below, and before using the diff functions, the two architecture objects are generated beforehand, as shown in the following script.
+
+```python
+from caret_analyze import Architecture
+left_arch = Architecture('yaml', 'old_architecture.yaml')
+right_arch = Architecture('yaml', 'new_architecture.yaml')
+
+```
+
+### diff_node_names()
+
+An example of the use of `diff_node_names()` is shown in `#sample_1`.
+
+<!-- Suppose arch1 has two nodes 'node1' and 'node2'. arch2 has three nodes 'node1', 'node3' and 'node4'.
+The type of the return value is `Tuple[Tuple[str, ...] , Tuple[str, ...]]` -->
+
+```python
+#sample_1
+
+#get node names that are only in old and new respectively.
+left_only_node_names, right_only_node_names = Architecture.diff_node_names(left_arch, right_arch)
+
+print(left_only_node_names)
+print(right_only_node_names)
+
+#result
+#('node2',)
+#('node3', 'node4')
+
+```
+
+In this setup, the `left_arch` has two nodes: `node1` and `node2`, and the `right_arch` has three nodes: `node1`, `node3` and `node4`. Therefore, the node name that `left_arch` and `right_arch` have in common is `node1`.
```

Comment:
```suggestion
In this setup, the `left_arch` has two nodes: `node1` and `node2`, and the `right_arch` has three nodes: `node1`, `node3` and `node4`. 
The `left_arch` and `right_arch` have `node1` as common node.
```

Reply:
Don't need to apply this modification.
Please apply [here](https://github.com/tier4/CARET_doc/pull/168#pullrequestreview-1347730340).